src/ucs/sys/sys.c

```diff
@@ -330,26 +331,51 @@ size_t ucs_get_page_size()
     return page_size;
 }
 
-size_t ucs_get_huge_page_size()
+size_t ucs_get_meminfo_entry(const char* pattern)
 {
-    static size_t huge_page_size = 0;
     char buf[256];
-    int size_kb;
+    char final_pattern[80];
+    int val = 0;
+    size_t val_b = 0;
     FILE *f;
 
-    /* Cache the huge page size value */
-    if (huge_page_size == 0) {
-        f = fopen("/proc/meminfo", "r");
-        if (f != NULL) {
-            while (fgets(buf, sizeof(buf), f)) {
-                if (sscanf(buf, "Hugepagesize:       %d kB", &size_kb) == 1) {
-                    huge_page_size = size_kb * 1024;
-                    break;
-                }
+    f = fopen("/proc/meminfo", "r");
+    if (f != NULL) {
+        snprintf(final_pattern, sizeof(final_pattern), "%s: %s", pattern,
+                 "%d kB");
+        while (fgets(buf, sizeof(buf), f)) {
+            if (sscanf(buf, final_pattern, &val) == 1) {
+                val_b = val * 1024ull;
+                break;
             }
-            fclose(f);
         }
+        fclose(f);
+    }
+
+    return val_b;
+}
+
+size_t ucs_get_memfree_size()
+{
+    size_t mem_free;
 
+    mem_free = ucs_get_meminfo_entry("MemFree");
+    if (mem_free == 0) {
+        mem_free = UCS_DEFAULT_MEM_FREE;
+        ucs_warn("cannot determine free mem size, using default: %zu",
```

Comment:
need to make it use_info. unlike ucs_get_huge_page_size(), it would be printed all the time (not only once) and will cause tests to fail because it's a warning